pkg/controlplane/reconcilers/lease.go

```diff
@@ -146,16 +147,14 @@ func NewLeases(config *storagebackend.ConfigForResource, baseKey string, leaseTi
 type leaseEndpointReconciler struct {
 	epAdapter             EndpointsAdapter
 	masterLeases          Leases
-	stopReconcilingCalled bool
-	reconcilingLock       sync.Mutex
+	stopReconcilingCalled atomic.Bool
```

Comment:
this is ok, this solves the problem of handling termination inside the `doReconcile` method